src/main/java/me/karakelley/http/handlers/Router.java

```diff
@@ -0,0 +1,30 @@
+package me.karakelley.http.handlers;
+
+import me.karakelley.http.Request;
+import me.karakelley.http.Response;
+
+import java.util.*;
+
+public class Router implements Handler {
+  private final Map<String, HashMap<List<String>, Handler>> mappings = new HashMap<>();
```

Comment:
This is an interesting data structure—I'll be interested in seeing how it works out. Kinda hard to tell prior to handling any non-GET requests.